src/test/java/org/bookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,241 @@
+    @DisplayName("Getting book by id")
+    @WithMockUser(username = "admin", authorities = {"ADMIN", "USER"})
+    void getBookById_existingBook_BookDto() throws Exception {
+        long id = 1;
+        MvcResult result = mockMvc.perform(get("/books/" + id))
+                .andExpect(status().isOk())
+                .andReturn();
+        BookDto actual = objectMapper.readValue(result.getResponse().getContentAsString(),
+                BookDto.class
+        );
+
+        assertThat(actual).isNotNull()
+                .hasFieldOrPropertyWithValue("id", 1L)
+                .hasFieldOrPropertyWithValue("title", "book-1")
+                .hasFieldOrPropertyWithValue("author", "Alice")
+                .hasFieldOrPropertyWithValue("isbn", "isbn-000001")
+                .hasFieldOrPropertyWithValue("price", new BigDecimal("200.00"))
+                .hasFieldOrPropertyWithValue("description", "descr_1")
+                .hasFieldOrPropertyWithValue("coverImage", "image_1");
+    }
+
+    @Test
+    @DisplayName("Updating book by id")
+    @WithMockUser(username = "admin", authorities = {"ADMIN", "USER"})
+    void updateBookById_existingBook_BookDto() throws Exception {
+        long id = 1;
+        CreateBookRequestDto request = new CreateBookRequestDto(
+                "title",
+                "author_changed",
+                "12345678900",
+                BigDecimal.TEN,
+                "descr",
+                "image_changed",
+                Set.of(1L)
+        );
+
+        MvcResult result = mockMvc.perform(put("/books/" + id)
+                        .content(objectMapper.writeValueAsBytes(request))
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        BookDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(),
+                BookDto.class
+        );
+
+        assertThat(actual)
+                .hasFieldOrPropertyWithValue("author", "author_changed")
+                .hasFieldOrPropertyWithValue("coverImage", "image_changed");
+    }
+
+    @Test
+    @DisplayName("Update non-existing book")
+    @WithMockUser(username = "admin", authorities = {"USER", "ADMIN"})
+    @Sql(scripts = "classpath:/scripts/books/delete-books.sql")
+    void updateBookById_nonExistingBook_exception() throws Exception {
+        long id = 1;
+        CreateBookRequestDto request = new CreateBookRequestDto(
+                "title",
+                "author_changed",
+                "12345678900",
+                BigDecimal.TEN,
+                "descr",
+                "image_changed",
+                Set.of(1L)
+        );
+
+        mockMvc.perform(put("/books/" + id)
+                        .content(objectMapper.writeValueAsBytes(request))
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isNotFound())
+                .andReturn();
```

Comment:
no need  to return

pls apply to all places, where your don't do assertions on response body